torchtune/modules/loss/kd_losses.py

```diff
@@ -50,6 +52,8 @@ def forward(
         prod_probs = torch.masked_fill(teacher_prob * student_logprob, inf_mask, 0)
         x = torch.sum(prod_probs, dim=-1).view(-1)
         mask = (labels != self.ignore_index).int()
+        if not normalize:
+            return -torch.sum(x * mask.view(-1), dim=0)
         if torch.sum(mask.view(-1), dim=0) == 0:
             return torch.tensor(0.0, device=x.device)
         return -torch.sum(x * mask.view(-1), dim=0) / torch.sum(mask.view(-1), dim=0)
```

Comment:
FWIW we have `masked_mean` which we use in other parts of the codebase for this https://github.com/pytorch/torchtune/blob/17ba37d281548e5d60215f741888066717ad5b3e/torchtune/rlhf/rewards.py#L101 but not hugely fussed about whether it's used here.